src/DynamoCoreWpf/ViewModels/Core/InPortViewModel.cs

```diff
@@ -321,7 +342,7 @@ private void SetupDefaultPortColorValues()
                 PortBorderBrushColor = PortBorderBrushColorKeepListStructure;
             }
             // Port has a default value, shows blue marker
-            else if (UsingDefaultValue && DefaultValueEnabled)
+            else if (UsingDefaultValue && DefaultValueEnabled || (isReturningDefaultValue && !isFunctionNode))
```

Comment:
I don't understand how you can prove anything about the default value from the return value of the output ports -

I think it's important to point out that the python node executes because it doesn't NEED its inputs to be satisfied to execute, not because it has default values, it doesn't have default values for its ports, nor do other explicit NodeModel nodes, like ColorRange etc. 

Consider an arbitrary function in a familiar language like javascript, you can call a function and not pass all the arguments, the function still executes.

I'm not yet getting how default values factor into the requirements.

Reply:
just saw your comment above @filipeotero regarding getting rid of `isReturningDefaultValue` - sounds good to me.

Reply:
I think, more strictly speaking, @filipeotero is trying to determine if a node's input ports are "satisfied" (not necessarily default) by inspecting the node output after it has run, and if it returns a non-`Function` value, he's assuming all ports to be satisfied, whereas if it returns a `Function`, for them to be unsatisfied (if not connected). @filipeotero can you confirm if I'm understanding this correctly.

Reply:
yes @aparajit-pratap, you are right.

Reply:
Correct @filipeotero @aparajit-pratap - it's about port satisfaction. This simply means the node will run as intended. On non NodeModel Nodes, this happens when you wire in or unwire a connector. Doesn't have to wait for node execution.